mongodump/barrier.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) MongoDB, Inc. 2014-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+package mongodump
+
+import (
+	"os"
+	"time"
+
+	"github.com/mongodb/mongo-tools/common/log"
+)
+
+// Wait until a file exists and can be opened for reading
+// This is used only for testing mongodump/mongorestore with resmoke
+// test infrastructure.  The tests will create the barrier file when they have
+// finshed writes to the source cluster.
+func waitForSourceWritesDoneBarrier(barrierName string) {
+	log.Logvf(log.Always, "waitForSourceWritesDoneBarrier: %s", barrierName)
```

Comment:
```suggestion
	log.Logvf(log.DebugHigh, "waitForSourceWritesDoneBarrier: %s", barrierName)
```

I'm not too familiar with how we do logging in the tools. But, should this be at the debug level since we won't use this flag in production?

Reply:
Can we update the log message so that it's clear that this log means we are entering the blocking state and the later one means we are still in the blocking state.

Reply:
Done and done

Reply:
Going to disagree with @ajayvijayakumar123: if this should never run in production this it's actually critical that it is as noisy as possible when running in production. Basically, if we hide the logs when this is running in production then we'll not be able to detect that something _which should never run in production_ is running in production until it breaks.

If possible, we should not have this code compile into production binaries so that it's actually impossible to run in production :).

Reply:
Ah yup good point. I agree

Reply:
Undone and undone, and I added a comment to explain the choice of log.Always.